docs/model-dev-guide/profiling.rst

```diff
@@ -0,0 +1,110 @@
+profiling, each targeting a different layer in a training job at various levels of detail:
+
+-  :ref:`Determined system metrics profiler <how-to-profiling-det-profiler>` collects general
+   system-level metrics and provides an overview of hardware usage during an experiment.
+-  :ref:`Native profiler integration <how-to-profiling-native-profilers>` enables model profiling in
+   training APIs that provides fine-grained metrics specific to your model.
+-  :ref:`Prometheus/Grafana integration <how-to-profiling-prom-grafana>` can be set up to track
+   detailed hardware metrics and monitor overall cluster health.
+
+.. _how-to-profiling:
+
+.. _how-to-profiling-det-profiler:
+
+*********************
+ Determined Profiler
+*********************
+
+Determined comes with a built-in profiler that provides out-of-the-box tracking for system-level
+metrics. System metrics are statistics around hardware usage, such as GPU utilization, disk usage,
+and network throughput.
+
+These metrics provide a general overview of resource usage during a training run, and can be useful
+for quickly identifying ineffective usages of computational resources. When the system metrics
+reported for an experiment do not match hardware expectations, that is a sign that the software may
+be able to be optimized to make better use of the hardware resources.
+
+The Determined profiler collects a set of system metrics throughout an experiment which can be
+visualized in the WebUI under the experiment's "Profiler" tab. It is supported for all training
+APIs, but is not enabled by default.
+
+Visit :ref:`core-profiler` to find out how to enable and configure the Determined profiler for your
+experiment.
+
+The following system metrics are tracked:
+
+-  *GPU utilization (percent)*: current utilization of a GPU device
+-  *GPU free memory (bytes)*: current amount of free memory available on a GPU device
+-  *Network throughput - sent (bytes/s)*: change rate of number of bytes sent system-wide
+-  *Network throughput (received)*: change rate of number of bytes received system-wide
+-  *Disk IOPS (bytes/s)*: change rate of number of read + writes system-wide
+-  *Disk throughput - reads (bytes/s)*: change rate of bytes read system-wide
+-  *Disk throughput - writes (bytes/s)*: change rate of bytes written system-wide
+-  *Host available memory (gigabytes)*: current amount of memory available (not including swap)
+   system-wide
+-  *CPU utilization (percent)*: current utilization of CPU cores, averaged across all cores in the
+   system
+
+For distributed training, these metrics are collected for every agent. The data is broken down by
+agent, and GPU metrics can be further broken down by GPU.
+
+.. note::
+
+   System Metrics record agent-level metrics, so when there are multiple experiments on the same
+   agent, it is difficult to analyze. It is recommended that profiling is done with only a single
+   experiment per agent.
```

Comment:
This could be a place to reference how this can be configured.

Reply:
this can't really be configured. i left this warning from the previous doc because it's still relevant, but it has to do with being aware that there could be other experiments using the same agent as your experiment.

Reply:
I think it would be frustrating to me to read this note and then be unable to figure out how to do the thing it recommends. Let's chat about it.

Reply:
removed this recommendation